js/tji-datasets-explore.js

```diff
@@ -181,11 +188,22 @@ TJIGroupByDoughnutChart.prototype.get_options_overrides = function() {
 
 TJIGroupByDoughnutChart.prototype.create_legend = function(keys) {
   var that = this;
+  var colormap = {};
+  _.map(keys, function(key, i) { colormap[key] = that.colors[i] });
+  var keys_sorted = _.sortBy(keys);
```

Comment:
1. I'd add a comment that this alphabetizes the legend
2. I'm not sure this is what we want to do. The colors, clockwise around the donut, might read red-blue-yellow-purple, but the legend could read yellow-blue-purple-red, which I think makes it harder to look up values, and especially hard if you're having a tough time distinguishing two colors and rely on the order to help you out.

Reply:
This is in conflict with ordering the pie pieces by size for color coherency and keeping the legend alphabetical per your previous request. We need to determine if the legend and the order of pie pieces should be coupled or not as you've suggested here. If so, then we need to determine if color coherency is more important than alphabetizing the legend.